### ISSUE TITLE
Use stack cradle if no project config + ghc not in PATH + stack in PATH

### ISSUE THREAD

**Issue by jneira** [haskell/haskell-language-server#692]
* Opening a single file requirer to have ghc in path
* However beginners using stack can do a `stack repl` to load those files (stack uses the global project configuration)
* So if we have stack in PATH but no GHC we could use stack to load those files as well
* Not sure if it can be done in `hie-bios` or `implicit-hie-cradle`

**Comment by Avi-D-coder** [comment]
I can take a look on Christmas.

**Comment by jneira** [comment]
Thinking in it twice, there exists the possibility of use cabal, as you can set in cabal global config the path to ghc without having it in the global `PATH`. Not sure if that is a common setup though.
Maybe we should check if `cabal exec ghc -- --version` works (and in a similar way `stack exec ghc --  --version`, maybe with `--skip-ghc-check` to avoid its download?) and proceed to use cabal or stack if ghc is set.

**Comment by Ailrun** [comment]
Awesome, this is exactly what I have hoped for a few weeks, but couldn't come up with some specification. Thank you for raising this issue!

**Comment by jneira** [comment]
I have to say that it arose in a conversation with a haskell beginner (@hieung1401 in the fp discord channel), who was using directly stack for fire a repl.
He had to create a stack project only to make hls work. But even experienced haskell developers work with standalone files so it can be handy for everybody.

**Comment by Ailrun** [comment]
Yes, true. I installed system GHC only to make HLS work for small scripts that I need to use from time to time. If HLS supports such a stack detection mechanism, it would allow much compact setting for Haskell dev.

**Comment by jneira** [comment]
I would say that cabal could not be as useful as stack cause you can configure the stack global project as you want, including dependencies, but afaik you cant do the same in the global cabal config, only set the path to ghc.
So in this case i would choose stack first if both tools are available.

**Comment by jneira** [comment]
@Avi-D-coder has you the opportunity to make some progress on this?

**Comment by Avi-D-coder** [comment]
I took a look, but I have not had time to finish.

**Comment by July541** [comment]
Hello, I am a student who would like to apply for GSoC. HLS is a very useful project for every developer. As a beneficiary, I hope to contribute to this project. As HLS is a complex project that beyond my ability, some technical guide may help me establish a feasible plan. I want to know where I can get some help with the details of the proposal.

**Comment by jneira** [comment]
it is a common issue among beginners so probably we should add it as konwn issue in troubleshooting section

**Comment by July541** [comment]
> it is a common issue among beginners so probably we should add it as konwn issue in troubleshooting section

I did really met some students with this problem, especially some of them following the instruction from the class but not installed correctly(they thought they had made it), which cause HLS unable to work.

**Comment by ruifengx** [comment]
It seems using the following explicit cradle works:
```yaml
cradle:
  stack:
```
Save this as `hie.yaml` into the directory, and just use open VSCode without custom `PATH`. I assumed this configuration should be able to force `hls` to call `stack`, and it works pretty well.
